src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3681,6 +3681,12 @@ module EstablishTypeDefinitionCores =
                                   
                                   for slot in slots do 
                                       yield mkLocalValRef slot ]
+                            
+                        if hasAbstractAttr then
```

Comment:
Hmm. I guess #17044 should really have been about classes in general. This also yields an internal error:

```fsharp
#nowarn "3535"

type A () =
    static abstract M : unit -> unit
```

```
error FS0193: internal error: Unable to load one or more of the requested types.
Abstract method in non-abstract class.
```

I guess we don't really need to check for structs, since they already give an error for _any_ abstract members, static or not.

Reply:
Yeah make sense.